brew_view/controllers/request_list_api.py

```diff
@@ -401,24 +432,32 @@ def _get_requests(self, start, end):
 
         # Now we can construct the actual query parameters
         query_params = reduce(lambda x, y: x & y, search_params, Q())
+        query_set = query_set.filter(query_params)
 
-        # Further modify the query itself
+        # Further modify the QuerySet
         if overall_search:
-            query = query.search_text(overall_search)
+            query_set = query_set.search_text(overall_search)
 
         if order_by:
-            query = query.order_by(order_by)
+            query_set = query_set.order_by(order_by)
 
         # Marshmallow treats [] as 'serialize nothing' which is not what we
         # want, so translate to None
         if requested_fields:
-            query = query.only(*requested_fields)
+            query_set = query_set.only(*requested_fields)
         else:
             requested_fields = None
 
-        # Execute the query / count
-        requests = query.filter(query_params)
-        filtered_count = requests.count()
-
-        # Only return the correct slice of the QuerySet
-        return requests[start:end], filtered_count, requested_fields
+        # Mongo seems to prefer using only the ['parent', '<sort field>']
+        # index, even when also filtering. So we have to help it a bit.
+        real_hint = ['parent']
```

Comment:
By default, (i.e. even if they didn't specify an order_by) we want to use the `parent_created_at_index` right?